examples/regression/2-advanced-analysis/plot_timeseries_enbpi.py

```diff
@@ -0,0 +1,212 @@
+        scoring="neg_root_mean_squared_error",
+        random_state=random_state,
+        verbose=0,
+        n_jobs=-1,
+    )
+    cv_obj.fit(X_train, y_train)
+    model = cv_obj.best_estimator_
+else:
+    # Model: Random Forest previously optimized with a cross-validation
+    model = RandomForestRegressor(
+        max_depth=10, n_estimators=50, random_state=59
+    )
+
+# Estimate prediction intervals on test set with best estimator
+alpha = 0.05
+cv_mapietimeseries = BlockBootstrap(
+    n_resamplings=100, length=48, overlapping=True, random_state=59
+)
+
+mapie_enpbi = MapieTimeSeriesRegressor(
+    model,
+    method="enbpi",
+    cv=cv_mapietimeseries,
+    agg_function="mean",
+    n_jobs=-1,
+)
+
+print("EnbPI, with no partial_fit, width optimization")
+mapie_enpbi = mapie_enpbi.fit(X_train, y_train)
+y_pred_npfit_enbpi, y_pis_npfit_enbpi = mapie_enpbi.predict(
+    X_test, alpha=alpha, ensemble=True, optimize_beta=True
+)
+coverage_npfit_enbpi = regression_coverage_score(
+    y_test, y_pis_npfit_enbpi[:, 0, 0], y_pis_npfit_enbpi[:, 1, 0]
+)
+
+width_npfit_enbpi = regression_mean_width_score(
+    y_pis_npfit_enbpi[:, 1, 0], y_pis_npfit_enbpi[:, 0, 0]
+)
+
+print("EnbPI with partial_fit, width optimization")
+mapie_enpbi = mapie_enpbi.fit(X_train, y_train)
+y_pred_pfit_enbpi = np.zeros(y_pred_npfit_enbpi.shape)
+y_pis_pfit_enbpi = np.zeros(y_pis_npfit_enbpi.shape)
+
+step_size = 1
+(
+    y_pred_pfit_enbpi[:step_size],
+    y_pis_pfit_enbpi[:step_size, :, :],
+) = mapie_enpbi.predict(
+    X_test.iloc[:step_size, :], alpha=alpha, ensemble=True, optimize_beta=True
+)
+
+for step in range(step_size, len(X_test), step_size):
+    mapie_enpbi.partial_fit(
+        X_test.iloc[(step - step_size):step, :],
+        y_test.iloc[(step - step_size):step],
+    )
+    (
+        y_pred_pfit_enbpi[step:step + step_size],
+        y_pis_pfit_enbpi[step:step + step_size, :, :],
+    ) = mapie_enpbi.predict(
+        X_test.iloc[step:(step + step_size), :],
+        alpha=alpha,
+        ensemble=True,
+        optimize_beta=True,
+    )
+coverage_pfit_enbpi = regression_coverage_score(
+    y_test, y_pis_pfit_enbpi[:, 0, 0], y_pis_pfit_enbpi[:, 1, 0]
+)
+width_pfit_enbpi = regression_mean_width_score(
+    y_pis_pfit_enbpi[:, 1, 0], y_pis_pfit_enbpi[:, 0, 0]
+)
+
+# Print results
+print(
+    "Coverage / prediction interval width mean for MapieTimeSeriesRegressor: "
+    "\nEnbPI without any partial_fit:"
+    f"{coverage_npfit_enbpi :.3f}, {width_npfit_enbpi:.3f}"
+)
+print(
+    "Coverage / prediction interval width mean for MapieTimeSeriesRegressor: "
+    "\nEnbPI with partial_fit:"
+    f"{coverage_pfit_enbpi:.3f}, {width_pfit_enbpi:.3f}"
+)
+
+# Plot estimated prediction intervals on test set
+fig, (ax1, ax2) = plt.subplots(
+    nrows=2, ncols=1, figsize=(30, 25), sharey="row", sharex="col"
+)
+
+for ax in [ax1, ax2]:
+    ax.set_ylabel("Hourly demand (GW)")
+    ax.plot(demand_test.Demand, lw=2, label="Test data", c="C1")
+
+ax1.plot(
```

Comment:
Even with two approaches, we can include the plotting commands with a loop (see notebook for instance).